D3D12Bundles/D3D12Bundles.cs

```diff
@@ -187,45 +178,42 @@ struct Vertex {
         /// Render the scene
         /// </summary>
         public virtual void OnRender() {
-            using (var pe = new ProfilingEvent(mCommandQueue, "Render")) {
+            using (var pe = new ProfilingEvent(mGraphicsDevice.DirectCommandQueue.NativeCommandQueue, "Render")) {
                 // Record all the commands we need to render the scene into the command list.
-                PopulateCommandList(mCurrentFrameResource);
+                CompiledCommandList compiledCommandList = PopulateCommandList(mGraphicsDevice.CommandList, mCurrentFrameResource);
 
                 // Execute the command list.
-                mCommandQueue.ExecuteCommandList(mCommandList);
+                mGraphicsDevice.DirectCommandQueue.ExecuteCommandList(compiledCommandList);
             }
 
             // Present and update the frame index for the next frame.
-            mSwapChain.Present(1, 0).CheckError();
-            mFrameIndex = mSwapChain.CurrentBackBufferIndex;
+            mPresenter.Present();
 
             // Signal and increment the fence value.
-            mCurrentFrameResource.mFenceValue = mFenceValue;
-            mCommandQueue.Signal(mFence, mFenceValue).CheckError();
-            mFenceValue++;
+            mCurrentFrameResource.mFenceValue = mGraphicsDevice.DirectCommandQueue.AddSignal();
         }
 
         public virtual void OnDestroy() {
             // Ensure that the GPU is no longer referencing resources that are about to be
             // cleaned up by the destructor.
             {
-                var fence = mFenceValue;
-                var lastCompletedFence = mFence.CompletedValue;
-
-                // Signal and increment the fence value.
-                mCommandQueue.Signal(mFence, mFenceValue).CheckError();
-                mFenceValue++;
+                // Add signal to the command queue.
+                var fence = mGraphicsDevice.DirectCommandQueue.AddSignal();
 
                 // Wait until the previous frame is finished.
-                if (lastCompletedFence < fence) {
-                    mFence.SetEventOnCompletion(fence, mFenceEvent).CheckError();
-                    mFenceEvent.WaitOne();
-                }
+                mGraphicsDevice.DirectCommandQueue.WaitForSignal(fence);
```

Comment:
Oops, forgot to leave these comments before I merged the PR...

> Could achieve `L197` by constructing a new command list, and compiling it, sheduling execution, and waiting for it to complete?  Assuming it's possible to have empty command lists, that would achieve what is required here without any further changes to the DX12GE plumbing...
>
> But that is a bit heavy, since it would also create a command allocator and set descriptor heaps...
It would be more convenient to publicly expose manual signalling and waiting methods...
Or a single method that takes an optional delegate of work to do in the middle... although that's less flexible.
In this project, `Update` and `Render` coordinate using a fence so a single method would prevent that pattern.

In the end you see I opted for two methods, which is reusable in more scenarios than just this one.